dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityLiveMode.java

```diff
@@ -0,0 +1,200 @@
+                 */
+                Logger.debug(this.getClass(), "An EVERY_PAGE RuleEngine Action has committed the response.");
+                return;
+            }
+
+
+            User user = null;
+            HttpSession session = request.getSession(false);
+
+            if (session != null) {
+                user = (User) session.getAttribute(com.dotmarketing.util.WebKeys.CMS_USER);
+            }
+
+            Logger.debug(this.getClass(), "Page Permissions for URI=" + uri);
+
+
+
+            IHTMLPage htmlPage = VelocityUtil.getPage(id, langId, mode.showLive);
+
+
+            // Verify and handle the case for unauthorized access of this contentlet
+            Boolean unauthorized = CMSUrlUtil.getInstance().isUnauthorizedAndHandleError(htmlPage, uri, user, request, response);
+            if (unauthorized) {
+                return;
+            }
+
+            // Fire the page rules until we know we have permission.
+            RulesEngine.fireRules(request, response, htmlPage, Rule.FireOn.EVERY_PAGE);
+
+            Logger.debug(this.getClass(), "Recording the ClickStream");
+            if (Config.getBooleanProperty("ENABLE_CLICKSTREAM_TRACKING", false)) {
+                if (user != null) {
+                    UserProxy userProxy = com.dotmarketing.business.APILocator.getUserProxyAPI().getUserProxy(user,
+                            APILocator.getUserAPI().getSystemUser(), false);
+                    if (!userProxy.isNoclicktracking()) {
+                        ClickstreamFactory.addRequest((HttpServletRequest) request, ((HttpServletResponse) response), host);
+                    }
+                } else {
+                    ClickstreamFactory.addRequest((HttpServletRequest) request, ((HttpServletResponse) response), host);
+                }
+            }
+
+            // Begin page caching
+            String userId = (user != null) ? user.getUserId() : APILocator.getUserAPI().getAnonymousUser().getUserId();
+            String language = String.valueOf(langId);
+            String urlMap = (String) request.getAttribute(WebKeys.WIKI_CONTENTLET_INODE);
+            String queryString = request.getQueryString();
+            String persona = null;
+            Optional<Visitor> v = visitorAPI.getVisitor(request, false);
+            if (v.isPresent() && v.get().getPersona() != null) {
+                persona = v.get().getPersona().getKeyTag();
+            }
+
+
+            PageCacheParameters cacheParameters =
+                    new BlockPageCache.PageCacheParameters(userId, language, urlMap, queryString, persona);
+
+            boolean buildCache = false;
+            String key = VelocityUtil.getPageCacheKey(request, response);
+            if (key != null) {
+                String cachedPage = CacheLocator.getBlockPageCache().get(htmlPage, cacheParameters);
+                if (cachedPage == null || "refresh".equals(request.getParameter("dotcache"))
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "dotcache" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "refresh" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)